hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/SelectProcessorSupplier.java

```diff
@@ -151,4 +152,17 @@ public void readData(ObjectDataInput in) throws IOException {
         parameterPositions = in.readIntArray();
         dialectName = in.readString();
     }
+
+    private String isNumberTypeCheck(ResultSetMetaData metaData, String type, int column) throws SQLException {
+        if (type.equals("NUMBER")) {
+            int precision = metaData.getPrecision(column + 1);
+            int scale = metaData.getScale(column + 1);
+            if (precision == 63 && scale == -127) {
+                return "REAL";
+            } else if (precision == 38 && scale == 0) {
+                return "INT";
+            }
```

Comment:
The precision can have any value, if it has e.g. 16, we should map it to smallint, etc...